engine/src/ast/field_expr.rs

```diff
@@ -505,7 +516,13 @@ impl Expr for ComparisonExpr {
                 }
 
                 match op {
-                    OrderingOp::NotEqual => gen_ordering!(!=, true),
+                    OrderingOp::NotEqual => {
+                        if nil_not_equal_behavior {
+                            gen_ordering!(!=, true)
+                        } else {
+                            gen_ordering!(!=, false)
+                        }
+                    }
```

Comment:
Is it possible to just do this?
```suggestion
                    OrderingOp::NotEqual => gen_ordering!(!=, nil_not_equal_behavior),
```